---
title: "CCM Analysis Replicate 1"
author: "Benjamin Reisman"
date: "10/14/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



```{r readandtidyr}
aacountpaths <- list.files("gatk_analysis", pattern = "aaCounts", full.names = T)
#aacountpaths.plasmid <- list.files("gatk_analysis_plasmid", pattern = "bwa.aaCounts", full.names = T)
#aacountpaths.plasmid
#i <- aacountpaths[[1]]
library(here)
library(dplyr)
library(stringr)
library(tidyverse)
file.path("")
here('..')
aaCounts <- lapply(aacountpaths, function(i) {
  read_tsv(i) %>%
  mutate(file =  str_split(basename(i), "\\.")[[1]][1]) %>%
  mutate(residue = 1:n())
})  %>%
  bind_rows()

list.files("gatk_analysis")
aaCounts.plasmid <- lapply(aacountpaths.plasmid, function(i) {
  read_tsv(i) %>%
  mutate(file =  str_split(basename(i), "\\.")[[1]][1]) %>%
  mutate(residue = 1:n())
})  %>%
  bind_rows()
aaCounts.plasmid
valid.residue <- c('R253', 'L102', 'G101', 'R100', 'K55', 'M48', 'I44', 'I41',
                   'D450', 'S447', 'L446', 'E445', 'L441', 'I440', 'I437', 'D436',
                   'D452', 'F449', 'Q448', 'F446', 'A445')

wt.tbl <- tibble(chain = c(rep('ATP5C', 8), rep('ATP5B', 8), rep('ATP5A1', 5)), 
                 aa.residue = valid.residue) %>%
          mutate(residue = as.integer(substr(aa.residue, 2, 10))) 

allcounts.plasmid  <- 
  aaCounts.plasmid %>%
  mutate(file = gsub("_ATP5", "-ATP5", file)) %>%
  separate(file, c('drop1', 'drop2', "chain"), sep = "-") %>%
  dplyr::select(-drop1, -drop2) %>%
  gather("aa", "count", -c('chain', 'residue')) %>%
  mutate(sample = 'library') %>%
  dplyr::filter(count > 100) %>%
  group_by(sample, chain, residue) %>%
  add_tally() %>%
  mutate(WT = count == max(count), 
         frac = count/sum(count)) %>%
  left_join(wt.tbl) %>%
  filter(!is.na(aa.residue)) %>%
  ungroup() %>%
  dplyr::select(-count, -n)


allcounts.plasmid
allcounts  <- 
  aaCounts %>%
  mutate(file = gsub("_ATP5", "-ATP5", file)) %>%
  separate(file, c("sample", "compound", "chain"), sep = "-") %>%
  group_by(sample, compound, chain) %>%
  #filter(sample %in% c("WT","CCM1", "CCM2", "CCM3")) %>%
  gather("aa", "count", -c('sample', 'compound', 'chain', 'residue')) %>%
  dplyr::filter(count > 100) %>%
  group_by(sample, compound, chain, residue) %>%
  add_tally() %>%
  mutate(WT = count == max(count), 
         frac = count/sum(count)) %>%
  left_join(wt.tbl) %>%
  dplyr::filter(!is.na(aa.residue)) %>%
  ungroup() %>%
  dplyr::select(-n, -count)

parental <- allcounts %>%
  dplyr::filter(compound == "Parental") %>%
  rename(frac.parental = frac) %>%
  select(-compound, -WT) 

allodds <- 
  allcounts %>%
    full_join(parental) %>%
  mutate(odds = log10((frac)/(1-(frac)))) %>%
  mutate(odds.parental = log10((frac.parental)/(1-(frac.parental)))) %>%
  mutate(enrich = odds - odds.parental) %>%
  arrange(desc(residue)) %>%
  mutate(residue = as.factor(residue)) %>%
  mutate(residue = forcats::fct_rev(residue))

allodds %>%
  dplyr::filter(residue == 437, compound == "Ammo_3") %>%
  dplyr::filter(residue == "I")
allodds.wt <- 
  allodds %>%
  dplyr::filter(WT == TRUE) %>%
  rename(enrich.wt = enrich) %>%
  select(sample, compound, chain, residue, enrich.wt)


allodds.norm <- allodds %>%
  left_join(allodds.wt) %>% 
  mutate(enrich.norm = enrich - enrich.wt) 



variant.counts <- allodds.norm  %>%
  select(sample, compound, chain , residue, aa, aa.residue, enrich.norm) %>%
  group_by(sample, compound) %>%
  tally()


allodds.norm  %>%
  select(sample, compound, chain , residue, aa, aa.residue, enrich.norm) %>%
  spread(sample, enrich.norm) %>%
  mutate(CCM1 = replace_na(CCM1, -3)) %>%
  mutate(CCM2 = replace_na(CCM2, -3)) %>%
  mutate(CCM3 = replace_na(CCM3, -3)) %>%
  summary


allodds.norm  %>%
  select(sample, compound, chain , residue, aa, aa.residue, enrich) %>%
  spread(sample, enrich) %>%
  mutate(CCM1 = replace_na(CCM1, -3)) %>%
  mutate(CCM2 = replace_na(CCM2, -3)) %>%
  mutate(CCM3 = replace_na(CCM3, -3)) %>%
  ggplot(aes(x=CCM1, y = CCM2, col = chain)) + 
  scale_x_continuous(limits = c(-5, 5)) + 
  scale_y_continuous(limits = c(-5, 5)) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  coord_fixed() + 
  geom_point(size = 0.5) + 
  facet_wrap(compound~., nrow = 2) + 
  theme_bw() + 
  theme(text = element_text(size= 8))

ggsave('ccm1_v_cc2_enrich.png', width = 7, height = 5, units = "in", dpi = 300)

ammo3 %>%
  dplyr::filter(chain == "ATP5B", residue == 436)
ammo3_data %>%
  left_join(ammo3_wt) %>% 
  mutate(enrich.norm = enrich - enrich.wt) %>%
  ggplot(aes(x=enrich.wt)) + 
  geom_histogram()


# parental %>%
#   left_join(ammo3) %>%
#   mutate(enrich = frac.ammo3/frac.parental) %>%
#   dplyr::filter(WT == FALSE) %>%
#   select(-frac.parental, -frac.ammo3) %>%
#   group_by(chain, residue, aa) %>%
#   spread(sample, enrich) %>%
#   ggplot(aes(x=CCM1, y = CCM2)) +
#   geom_point() + 
#   scale_x_log10() +
#   scale_y_log10() +
#   geom_hline(yintercept = 1, linetype = 2) + 
#   geom_vline(xintercept = 1, linetype = 2) + 
#   theme_bw() + 
#   theme(panel.grid = element_blank(), strip.background = element_rect(fill = NA)) + 
#   facet_grid(chain~.) + 
#   coord_fixed()
# 
# ggsave('ammoa3_enrichment_ccm1_ccm2.png', width = 4, height = 8, units = "in", dpi = 300)
ammo3_data.norm %>%
  ggplot(aes(x=enrich.norm)) + 
  geom_histogram()

list.files(pattern = "csv")
aa_groupings <- read_csv("aa_groupings.csv")
table(allcounts$sample)


res.order.0 <- 
  allodds %>%
  dplyr::filter(WT == TRUE) %>%
  dplyr::filter(compound == "Parental") %>%
  mutate(residue = as.numeric(as.character(residue))) %>%
  mutate(aa.residue = paste0(aa, residue)) %>%
#  mutate(aa.residue = factor(aa.residue)) %>%
  ungroup() %>%
  select(chain, residue, aa.residue)

res.order.0
res.order <- 
  res.order.0 %>%
  group_by(chain, residue) %>%
  slice(1) %>%
  pull(aa.residue)

res.order.tbl <- 
  res.order.0 %>%
  mutate(aa.residue = factor(aa.residue, levels = rev(res.order))) %>%
  mutate(residue = as.character(residue)) %>%
  group_by(chain, residue, aa.residue) %>%
  slice(1)
#res.order.tbl
allodds.norm %>% 
  dplyr::filter(chain == "ATP5B", residue == 436)


allodds.wt %>%
  dplyr::filter(residue == 437)
allcounts %>%
    dplyr::filter(compound == "Ammo_3") %>%
    dplyr::filter(chain == "ATP5B", residue == 437, aa == "I") 

library(scico)
WT.alleles
WT.alleles <- 
  allodds.norm %>%
  dplyr::filter(aa == substr(aa.residue, 1,1)) %>%
  group_by(aa.residue, chain, aa) %>%
  slice(1) %>%
  select(aa.residue, chain, aa) %>%
   inner_join(aa_groupings) %>%
  mutate(group = factor(
    group,
    levels  = c(
      "Hydrophobic",
      "Disruptive",
      "Hydrophilic",
      "Positive",
      "Negative"
    )
  )) %>%
  left_join(res.order.tbl) %>%
  dplyr::filter(!is.na(aa.residue)) %>%
  dplyr::filter(aa.residue %in% valid.residue)

WT.alleles

allodds.norm %>%
  group_by(chain, sample, compound) %>%
  left_join(res.order.tbl) %>%
  dplyr::filter(!is.na(aa.residue)) %>%
  filter(aa.residue %in% valid.residue) %>%
  left_join(res.order.tbl) %>%
  mutate(mut = aa != substr(aa.residue, 1, 1)) %>%
  summarise(mut.frac = sum(frac*as.numeric(mut))) %>%
  filter(mut.frac > 1) %>%
  ggplot(aes(x=mut.frac)) + 
  geom_histogram()
  
  
allodds.norm %>%
  group_by(chain, sample, compound) %>%
  left_join(res.order.tbl) %>%
  filter(!is.na(aa.residue)) %>%
  filter(aa.residue %in% valid.residue) %>%
  left_join(res.order.tbl) %>%
    mutate(mut = aa != substr(aa.residue, 1, 1)) %>%
  filter(compound == "Apo_1000", sample == "CCM1") %>%
  filter(mut == T, chain == "ATP5C") %>%
  arrange(desc(frac))

allodds.norm %>%
  filter(complete.cases(.)) %>%
  filter(enrich >= 0 | WT == TRUE) %>%
  group_by(chain, compound, residue, aa) %>%
  add_tally() %>%
  filter(n > 1) %>%
  group_by(chain, compound, residue, aa) %>%
  summarise(enrich = mean(enrich)) %>%
  spread(compound, enrich) %>%
  mutate(level = "") %>%
  mutate(level = if_else(!is.na(Parental), "Parental","")) %>%
  mutate(level = if_else(Ammo_3 > 1, "Ammocidin 3",level, missing = level)) %>%
  mutate(level = if_else(Ammo_30 > 1, "Ammocidin 30",level, missing = level)) %>%
  mutate(level = if_else(Ammo_300 > 1, "Ammocidin 300",level, missing = level)) %>%
  inner_join(aa_groupings) %>%
  mutate(group = factor(
    group,
    levels  = c(
      "Hydrophobic",
      "Disruptive",
      "Hydrophilic",
      "Positive",
      "Negative"
    )
  )) %>%
  left_join(res.order.tbl) %>%
  filter(!is.na(aa.residue)) %>%
  filter(aa.residue %in% valid.residue) %>%
  mutate(level = if_else(aa == substr(aa.residue, 1, 1), "Parental", level)) %>%
  mutate(level = factor(level, levels = c("Parental", "Ammocidin 3", "Ammocidin 30", "Ammocidin 300"))) %>%
  ggplot(aes(y = aa.residue, x = aa, fill = level)) + 
  geom_tile(color = "white", size = 1) + 
  scale_fill_brewer(type = "seq", palette = 12) + 
    geom_point(
    data = WT.alleles,
    inherit.aes = F,
    aes(y = aa.residue, x = aa)
  ) +
  facet_grid(chain ~ group, scales = "free", space = "free") + 
  theme_bw() + 
  theme(panel.grid = element_blank())

ggsave('enrichment_all_10fold.pdf', width = 8, height = 6, units = "in", dpi = 300)
#allodds.norm
allodds.norm %>% 
  ungroup() %>%
  mutate(enrich.norm = if_else(is.nan(enrich.norm), 0, enrich.norm)) %>%
 # filter(complete.cases(.)) %>%
  group_by(compound, chain , residue, aa) %>%
 # filter(compound == "Apo_100") %>%
  
 # filter(sample == "CCM2") %>%
  filter(compound == "Apo_1000") %>%
#  filter(aa == "R")
  summarise(enrich = mean(enrich, na.rm = T), 
            frac = mean(log10(frac) - mean(log10(frac.parental)))) %>%
  inner_join(aa_groupings) %>%
  mutate(group = factor(
    group,
    levels  = c(
      "Hydrophobic",
      "Disruptive",
      "Hydrophilic",
      "Positive",
      "Negative"
    )
  )) %>%
#  filter(enrich > -0.5) %>%
  #  filter(compound == "Ammo_3", residue == 102) 
  left_join(res.order.tbl) %>%
#  select(chain, residue, aa, enrich, group, aa.residue) %>%
  filter(!is.na(aa.residue)) %>%
  filter(aa.residue %in% valid.residue) %>%
  mutate(enrich = if_else(enrich == Inf, 0, enrich)) %>%
  #filter(residue == 101, aa == F)
  ggplot(aes(y = aa.residue, x = aa, fill = enrich)) +
  geom_tile(color = "white", size = 1) +
  facet_grid(chain ~ group, scales = "free", space = "free") +
  #scale_fill_viridis_c() + 
  scale_fill_scico(
    palette = 'cork',
    na.value = "grey50",
    limits = c(-5, 5),
    name = "log10\nEnrichment\nAmmo 30 nm"
   # name = "log10\nEnrichment\nApo 100 nM"
  ) +
#   scale_fill_scico(palette = 'bilbao') +
  geom_point(
    data = WT.alleles,
    inherit.aes = F,
    aes(y = aa.residue, x = aa)
  ) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        strip.background = element_rect(fill = NA), 
        text = element_text(color = 'black'), 
        axis.text = element_text(color = 'black')) +
  scale_x_discrete(name = "a.a.")
ggsave('apo1000_enrichment_norm.png', width = 8, height = 6, units = "in", dpi = 300)



allodds.norm %>%
  ungroup() %>%
    filter(complete.cases(.)) %>%
  group_by(compound, chain , residue, aa) %>%
  filter(compound == "Ammo_3" | compound == "Apo_100") %>%
 # filter(sample == "CCM1") %>%
#  filter(compound == "Ammo_3") %>%
  summarise(enrich.norm = mean(enrich.norm, na.rm = T)) %>%
  spread(compound, enrich.norm) %>%
  mutate(Ammo_3 = replace_na(Ammo_3, -4)) %>%
  mutate(Apo_100 = replace_na(Apo_100, -4)) %>%
  filter(Apo_100 > 1) %>%
  filter(Ammo_3 < 1)

  ggplot(aes(x=Ammo_3, y = Apo_100, col = chain)) + 
  scale_x_continuous(limits = c(-5, 5)) + 
  scale_y_continuous(limits = c(-5, 5)) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  geom_point() + 
  theme_bw() + 
  coord_fixed()

ggsave('apoA100_ammo3.png', width = 5, height = 5, units = "in", dpi = 300)




  inner_join(aa_groupings) %>%
  mutate(group = factor(
    group,
    levels  = c(
      "Hydrophobic",
      "Disruptive",
      "Hydrophilic",
      "Positive",
      "Negative"
    )
  )) %>%
  left_join(res.order.tbl) %>%
  select(chain, residue, aa, enrich.norm, group, aa.residue) %>%
  filter(!is.na(aa.residue)) %>%
  filter(aa.residue %in% valid.residue) %>%
  ggplot(aes(y = aa.residue, x = aa, fill = enrich.norm)) +
  geom_tile(color = "white", size = 1) +
  facet_grid(chain ~ group, scales = "free", space = "free") +
  scale_fill_scico(
    palette = 'cork',
    na.value = "grey97",
    limits = c(-5, 5),
   # name = "log10\nEnrichment\nApo 100 nm"
    name = "log10\nEnrichment\nAmmo 3 nM"
  ) +
  geom_point(
    data = WT.alleles,
    inherit.aes = F,
    aes(y = aa.residue, x = aa)
  ) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        strip.background = element_rect(fill = NA), 
        text = element_text(color = 'black'), 
        axis.text = element_text(color = 'black')) +
  scale_x_discrete(name = "a.a.")


allodds.norm %>%
  filter(aa != "X") %>%
  filter(compound == "Apo_100") %>%
  group_by(chain, residue, aa) %>%
  filter(sample == "CCM1") %>%
  #summarise(enrich.norm = mean(enrich.norm, na.rm = T)) %>%
  left_join(aa_groupings) %>%
  mutate(group = factor(
    group,
    levels  = c(
      "Hydrophobic",
      "Disruptive",
      "Hydrophilic",
      "Positive",
      "Negative"
    )
  )) %>%
  left_join(res.order.tbl) %>%
  select(chain, residue, aa, enrich.norm, group, aa.residue) %>%
  filter(!is.na(aa.residue)) %>%
  filter(aa.residue %in% valid.residue) %>%
ggplot(aes(y = aa.residue, x = aa, fill = enrich.norm)) +
  geom_tile(color = "white", size = 1) +
  facet_grid(chain ~ group, scales = "free", space = "free") +
  scale_fill_scico(
    palette = 'cork',
    na.value = "grey90",
    limits = c(-5, 5),
    name = "log10\nEnrichment\nAmmo 3 nm"
  ) +
  geom_point(
    data = WT.alleles,
    inherit.aes = F,
    aes(y = aa.residue, x = aa)
  ) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        strip.background = element_rect(fill = NA), 
        text = element_text(color = 'black'), 
        axis.text = element_text(color = 'black')) +
  scale_x_discrete(name = "a.a.") + 
  scale_y_discrete(breaks = res.order.tbl$aa.residue, name = "residue") + 
  labs(title = "DMS1-3")

ggsave('ammoa3_enrichment_dms1-3.png', width = 8, height = 6, units = "in", dpi = 300)

  pull(residue)

parental %>%
  left_join(vehicle) %>%
  mutate(enrich = frac.vehicle/frac.parental) %>%
  group_by(chain, residue, aa)# %>%
  summarise(enrich = mean(enrich, na.rm = T)) %>%
  ungroup() %>%
  arrange(residue) %>%
  mutate(residue = as.factor(residue)) %>%
  mutate(residue = forcats::fct_rev(residue)) %>%
  ggplot(aes(y=residue, x =aa, fill = enrich)) + 
  geom_tile(color= "white", size = 1) + 
  facet_wrap(chain~., scales = "free", nrow = 3) + 
  scale_fill_scico(palette = 'cork', trans = "log10", limits = c(0.01, 100), na.value = NA, name = "log10\nEnrichment\nVehicle", 
                   breaks = c(0.01, 0.1, 1, 10, 100), labels = c(0.01, 0.1, 1, 10, 100)) + 
  theme_bw() + 
  theme(panel.grid = element_blank(), strip.background = element_rect(fill = NA)) + 
  scale_x_discrete(name = "a.a.")

ggsave('vehicle_enrichment.png', width = 8, height = 8, units = "in", dpi = 300)

```




```{r}


allcounts.plasmid %>%
  filter(residue == 44) 
allcounts  %>%
  left_join(allcounts.plasmid %>%
  rename(frac.lib = frac) %>%
  select(-sample, -WT)) %>%
  filter(compound == "Parental") %>%
  mutate(odds = log10(frac/(1-frac)), 
         odds.lib = log10(frac.lib/(1-frac.lib))) %>%
  mutate(enrich = odds - odds.lib) %>%
   left_join(aa_groupings) %>%
  mutate(group = factor(
    group,
    levels  = c(
      "Hydrophobic",
      "Disruptive",
      "Hydrophilic",
      "Positive",
      "Negative"
    )
  )) %>%
  mutate(residue = as.character(residue)) %>%
  select(-aa.residue) %>%
  left_join(res.order.tbl) %>%
  filter(grepl("CCM", sample)) %>%
  group_by(compound, chain, residue, aa, aa.residue, group) %>%
  summarise(enrich = mean(enrich, na.rm = T))  %>%
  select(chain, residue, aa, enrich, group, aa.residue) %>%
  filter(!is.na(aa.residue)) %>%
  filter(aa.residue %in% valid.residue) %>%
  ggplot(aes(x=aa, y = aa.residue, fill = enrich)) + 
  geom_tile(color = 'white', size = 1) + 
  facet_grid(chain~group, scales = "free", space = 'free') + 
  scale_fill_scico(palette = 'cork', limits = c(-1.5, 1.5), na.value = NA) + 
  geom_point(
    data = WT.alleles,
    inherit.aes = F,
    aes(y = aa.residue, x = aa)
  ) +
  theme_bw() + 
  theme(panel.grid = element_blank())

ggsave('parental_v_library.png', width = 8, height = 6, units = "in", dpi = 300) 


allcounts  %>%
  left_join(allcounts.plasmid %>%
  rename(frac.lib = frac) %>%
  select(-sample, -WT)) %>%
  filter(compound == "Parental") %>%
  mutate(odds = log10(frac/(1-frac)), 
         odds.lib = log10(frac.lib/(1-frac.lib))) %>%
  mutate(enrich = odds - odds.lib) %>%
   left_join(aa_groupings) %>%
  mutate(group = factor(
    group,
    levels  = c(
      "Hydrophobic",
      "Disruptive",
      "Hydrophilic",
      "Positive",
      "Negative"
    )
  )) %>%
  mutate(residue = as.character(residue)) %>%
  select(-aa.residue) %>%
  left_join(res.order.tbl) %>%
  filter(grepl("CCM", sample)) %>%
  group_by(compound, chain, residue, aa, aa.residue, group) %>%
  ggplot(aes(x=odds.lib, y = odds, col = sample)) + 
  geom_point() + 
  geom_abline(slope = 1) + 
  facet_wrap(aa.residue~., nrow = 3) + 
  theme_bw() + 
  coord_fixed()

ggsave('lib_v_parental.png', width = 10, height = 7, units = "in", dpi = 300)
  summarise(enrich = mean(enrich, na.rm = T), 
            enrich = mean(enrich, na.rm = T))  %>%
  select(chain, residue, aa, enrich, group, aa.residue) %>%
  filter(!is.na(aa.residue)) %>%
  filter(aa.residue %in% valid.residue) 
allcounts  %>%
  left_join(allcounts.plasmid %>%
  rename(frac.lib = frac) %>%
  select(-sample, -WT)) %>%
  filter(sample == "CCM1", compound == "Parental") %>%
  mutate(odds = log10(frac/(1-frac)), 
         odds.lib = log10(frac.lib/(1-frac.lib))) %>%
  mutate(enrich = odds - odds.lib) %>%
  summary()
  filter(!WT) %>%
  ggplot(aes(x=odds.lib, y = odds)) + 
  geom_point() + 
  coord_fixed() 
allcounts.plasmid %>%
  bind_rows(allcounts) %>%
  filter(sample == 'library' | compound == 'parental') %>%
  filter(sample == 'library' | sample == 'CCM1') %>%
  select(-compound) %>%
  spread(aa.residue, frac)

?unite
```




```{r}
library(edgeR)
Pool <- factor(c(1,2,3,1,2,3))
Selection <- factor(c("Parental", "Parental", "Parental",
                      "Ammo30", "Ammo30", "Ammo30"))



rawcounts  <- aaCounts %>%
  mutate(file = gsub("_ATP5", "-ATP5", file)) %>%
  separate(file, c("sample", "compound", "chain"), sep = "-") %>%
  group_by(sample, compound, chain) %>%
  #filter(sample %in% c("WT","CCM1", "CCM2", "CCM3")) %>%
  gather("aa", "count", -c('sample', 'compound', 'chain', 'residue')) %>%
#  filter(count > 100) %>%
  group_by(sample, compound, chain, residue) %>%
  add_tally() %>%
  mutate(WT = count == max(count), 
         frac = count/sum(count)) %>%
  left_join(wt.tbl) %>%
  filter(!is.na(aa.residue)) %>%
  ungroup() %>%
  mutate(residue = paste0(aa.residue, aa)) %>%
  select(sample, compound, chain, residue, count) %>%
  unite(sample, sample, compound, sep = "-") %>%
  unite(feature, chain, residue, sep = "-")



library(ggrepel)
rawcount.mat <- rawcounts %>%
  filter(!grepl("WT", sample)) %>%
    filter(grepl("Parental", sample) | grepl("Ammo_30$", sample)) %>%
  spread(sample, count) %>%
  as.data.frame() %>%
  column_to_rownames("feature") %>%
  as.matrix()

treatment <- factor(c(2, 1, 2, 1, 2, 1))
sample <- factor(c(1,1, 2,2,3,3))
y <- DGEList(rawcount.mat, group = treatment)
design <- model.matrix(~treatment + sample)
y <- estimateDisp(y,design)
keep <- filterByExpr(y)
y <- y[keep,,keep.lib.sizes=FALSE]
y <- calcNormFactors(y)
fit <- glmQLFit(y,design)
qlf <- glmQLFTest(fit,coef=2)
fit <- glmFit(y,design)
lrt <- glmLRT(fit,coef=2)
lrt.ammo30 <- lrt

rawcount.mat <- rawcounts %>%
  filter(!grepl("WT", sample)) %>%
    filter(grepl("Parental", sample) | grepl("Apo_1000$", sample)) %>%
  spread(sample, count) %>%
  as.data.frame() %>%
  column_to_rownames("feature") %>%
  as.matrix()

treatment <- factor(c(2, 1, 2, 1, 2, 1))
sample <- factor(c(1,1, 2,2,3,3))
y <- DGEList(rawcount.mat, group = treatment)
design <- model.matrix(~treatment + sample)
y <- estimateDisp(y,design)
keep <- filterByExpr(y)
y <- y[keep,,keep.lib.sizes=FALSE]
y <- calcNormFactors(y)
fit <- glmQLFit(y,design)
qlf <- glmQLFTest(fit,coef=2)
fit <- glmFit(y,design)
lrt <- glmLRT(fit,coef=2)
lrt.apo1000 <- lrt


lrt.ammo30.tbl <- 
  lrt.ammo30$table %>%
  as.data.frame() %>%
  rownames_to_column("feature") %>%
  as_tibble() %>%
  rename(logFC.ammo30 = logFC, 
         pvalue.ammo30 = PValue) %>%
  select(feature, logFC.ammo30, pvalue.ammo30)

lrt.apo1000.tbl <- 
  lrt.apo1000$table %>%
  as.data.frame() %>%
  rownames_to_column("feature") %>%
  as_tibble() %>%
  rename(logFC.apo1000 = logFC, 
         pvalue.apo1000 = PValue) %>%
  select(feature, logFC.apo1000, pvalue.apo1000)

left_join(lrt.ammo30.tbl, lrt.apo1000.tbl) %>%
  mutate(signif = pvalue.ammo30 < 0.05 & pvalue.apo1000 < 0.05) %>%
  ggplot(aes(x=logFC.ammo30, y = logFC.apo1000, col = signif)) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  geom_point() +
  scale_color_manual(values = c("grey50", "green3"), guide = F) + 
  theme_bw() + 
  theme(panel.grid = element_blank())

left_join(lrt.ammo30.tbl, lrt.apo1000.tbl) %>%
  filter(pvalue.ammo30 < 0.05, 
         pvalue.apo1000 < 0.05)
#hist(lrt.apo1000.tbl$pvalue.apo1000)
volcano_data <- lrt.ammo30$table %>%
  as.data.frame() %>%
  rownames_to_column("feature") %>%
  as_tibble()
volcano_highlight <- volcano_data %>%
  filter(logFC > 5, 
         PValue < (0.05/420))

volcano_data %>%
  ggplot(aes(x = logFC, y = -log10(PValue))) + 
  geom_point() + 
  theme_classic() + 
  geom_text_repel(aes(label = feature), data = volcano_highlight) + 
  labs(x = "log2FC Ammocidin 30 nM") + 
  geom_hline(yintercept = -log10(0.05/420), linetype = 2)

ggsave("volcano_Ammo_30.png", width = 6, height = 6, units = "in", dpi = 300)
topTags(qlf)


topTags(lrt)
plotBCV(y)
plotMD(qlf)

ngenes <- 1000
nsamples <- 4
Counts <- matrix(rnbinom(ngenes*nsamples,mu=5,size=2),ngenes,nsamples)
rownames(Counts) <- 1:ngenes
y <- DGEList(counts=Counts, group=rep(1:2,each=2))
y
y
colnames(y)
y$samples
y$genes <- data.frame(Symbol=paste0("Gene",1:ngenes))
show(y)

```
## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
